Implement some methods for Set

deep110 opened the pull request.

deep110 commented:
@cflee @eliasdorneles there are many tests that are passing but pytest is not giving unexpected success, so should I remove that tests from test_set or let it be that way. 
Other than that this PR is ready for review.

deep110 commented:
@cflee @eliasdorneles Hi, does anything more need to be done in this PR.
Thanks!!

freakboy3742 commented on python/common/org/python/types/Set.java:
```diff
@@ -416,9 +455,12 @@ public boolean __setattr_null(java.lang.String name, org.python.Object value) {
-        set.retainAll(((Set) other).value);
-        return new Set(set);
+        try {
+            org.python.types.Set otherSet = new org.python.types.Set(new org.python.Object[] {other}, null);
```
It would be a lot more efficient to check whether `other` is already a set - and if it is, pass it unchanged to `__and__`, rather than universally duplicating it.

cflee commented:
@deep110 I think it'd be better to only mark the relevant tests as now implemented/passing, so that we don't have many other unrelated changes in this PR – thanks!

deep110 commented:
@freakboy3742 Thanks for the review. 
I have added type check for the set, and also updated difference method to take any iterable as mentioned in docs, I have added test to ensure its working.
